table/table.go

```diff
@@ -102,6 +93,15 @@ type Table interface {
 	// DeletableIndices returns delete-only, write-only and public indices of the table.
 	DeletableIndices() []Index
 
+	// Allocator returns Allocator.
+	Allocator(ctx sessionctx.Context) autoid.Allocator
+
+	// Meta returns TableInfo.
+	Meta() *model.TableInfo
+}
+
+// RegionKeyProvider defines some operations to locate the keys.
+type RegionKeyProvider interface {
```

Comment:
Why name it `RegionKey`?

Reply:
`Table` has nothing to do with region.

Reply:
We have the following choice:
1. Leave the `Table` interface unchanged.
2. Or, make it smaller, easier to composition.

`RegionKeyProvider` + `Common` could derive all the other methods in the `Table` interface, I tend to compose them.
If "Table has nothing to do with region", we can rename `Common` to `Table`, but we still need `RegionKeyProvider` to make all table related things functional.

Reply:
You can view `region` as a typo...
It's a concept related to `kv.Key` computation.